### DISCUSSION TITLE
Block with custom Dom Components type didn't save style applied

### DISCUSSION THREAD

**Discussion by ibnuhabibie** [GrapesJS/grapesjs#5029] [Q&A]
Hi, i am trying to create custom plugin for bootstrap icon. This is simple 1 block with custom domComponent type.
When block got double clicked, icon picker modal will popup. Then user can pick the icon they want.

The problem is when i try to add font-size or color, the style is applied to icon but when i change to other icon using class,
the style got reseted.

this is how i change block class.

```js
handleClick(e) {
      let selected = editor.getSelected();

      if (e.target.tagName == "SPAN") {
        var className = "bi bi-" + e.target.innerText;
      } else if (e.target.tagName == "I") {
        var className = `bi ${e.target.className}`;
      } else {
        var className = `bi ${e.target.firstChild.className}`;
      }

      selected.setClass(className);
      editor.Modal.close();
},
```

Demo link: https://codepen.io/ibnuhabibie/pen/WNgVRrj

**Comment by artf** [comment]
I guess you want to enable [this option](https://grapesjs.com/docs/modules/Selectors.html#component-first-selectors)

**Reply by ibnuhabibie** [reply]
I notice with this option, grapesjs will add unique id to each component inside canvas. what if i want to change all style with class `.footer-icon`. I've checked the css is updated on `#gjs-css-rules` but the component not updating.
